examples/demo_dlmbl/solution.py

```diff
@@ -389,80 +531,84 @@ def log_batch_jupyter(batch):
 
 # Dictionary that specifies key parameters of the model.
-phase2fluor_config = {
-    "architecture": "2D",
-    "num_filters": [24, 48, 96, 192, 384],
-    "in_channels": 1,
-    "out_channels": 2,
-    "residual": True,
-    "dropout": 0.1,  # dropout randomly turns off weights to avoid overfitting of the model to data.
-    "task": "reg",  # reg = regression task.
-}
+
+phase2fluor_config = dict(
+    in_channels=1,
+    out_channels=2,
+    encoder_blocks=[3, 3, 9, 3],
+    dims=[96, 192, 384, 768],
+    decoder_conv_blocks=2,
+    stem_kernel_size=(1, 2, 2),
+    in_stack_depth=1,
+    pretraining=False,
+)
 
 phase2fluor_model = VSUNet(
+    architecture="UNeXt2_2D",  # 2D UNeXt2 architecture
     model_config=phase2fluor_config.copy(),
-    batch_size=BATCH_SIZE,
-    loss_function=torch.nn.functional.l1_loss,
+    loss_function=MixedLoss(l1_alpha=0.5, l2_alpha=0.0, ms_dssim_alpha=0.5),
     schedule="WarmupCosine",
-    log_num_samples=5,  # Number of samples from each batch to log to tensorboard.
-    example_input_yx_shape=YX_PATCH_SIZE,
+    lr=2e-4,
+    log_batches_per_epoch=5,  # Number of samples from each batch to log to tensorboard.
+    freeze_encoder=False,
 )
 
-
 # %% [markdown]
 """
 ### Instantiate data module and trainer, test that we are setup to launch training.
 """
 # %%
+source_channel = ["Phase3D"]
+target_channel = ["Nucl", "Mem"]
 # Setup the data module.
-phase2fluor_data = HCSDataModule(
+phase2fluor_2D_data = HCSDataModule(
     data_path,
-    source_channel="Phase",
-    target_channel=["Membrane", "Nuclei"],
+    architecture="UNeXt2_2D",
+    source_channel=source_channel,
+    target_channel=target_channel,
     z_window_size=1,
     split_ratio=0.8,
     batch_size=BATCH_SIZE,
     num_workers=8,
-    architecture="2D",
     yx_patch_size=YX_PATCH_SIZE,
     augmentations=augmentations,
+    normalizations=normalizations,
 )
-phase2fluor_data.setup("fit")
+phase2fluor_2D_data.setup("fit")
 # fast_dev_run runs a single batch of data through the model to check for errors.
 trainer = VSTrainer(accelerator="gpu", devices=[GPU_ID], fast_dev_run=True)
 
 # trainer class takes the model and the data module as inputs.
-trainer.fit(phase2fluor_model, datamodule=phase2fluor_data)
+trainer.fit(phase2fluor_model, datamodule=phase2fluor_2D_data)
 
 
 # %% [markdown]
 # ## View model graph.
 #
-# PyTorch uses dynamic graphs under the hood. The graphs are constructed on the fly. This is in contrast to TensorFlow, where the graph is constructed before the training loop and remains static. In other words, the graph of the network can change with every forward pass. Therefore, we need to supply an input tensor to construct the graph. The input tensor can be a random tensor of the correct shape and type. We can also supply a real image from the dataset. The latter is more useful for debugging.
+# PyTorch uses dynamic graphs under the hood.
+# The graphs are constructed on the fly.
+# This is in contrast to TensorFlow,
```

Comment:
Is this still true for TF 2.x? https://www.tensorflow.org/guide/migrate/tf1_vs_tf2#eager_execution